### PR TITLE
[Feature] Support FLAVR Net

### PR DESCRIPTION
**Pull request by Yshuo-Li**

### CODE REVIEW

File: mmedit/models/backbones/vfi_backbones/flavr_net.py [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+                stride=1,
+                padding=0))
+
+    def forward(self, images: torch.Tensor):
+        # from [b, t, c, h, w] to [b, c, d, h, w], where t==d
+        images = images.permute((0, 2, 1, 3, 4))
+
+        # Batch mean normalization works slightly better than global mean
+        # normalization, Refer to https://github.com/myungsub/CAIN
+        mean_ = images.mean(
```
Comment (wangruohui) [inline mmedit/models/backbones/vfi_backbones/flavr_net.py:82]: image,mean(dim=(2,3,4), True)

File: mmedit/models/backbones/vfi_backbones/flavr_net.py [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+                3, keepdim=True).mean(
+                    4, keepdim=True)
+        images = images - mean_
+
+        xs = self.encoder(images)
+
+        dx_out = self.decoder(xs)
+
+        out = self.feature_fuse(dx_out)
+        out = self.conv_last(out)
```
Comment (wangruohui) [inline mmedit/models/backbones/vfi_backbones/flavr_net.py:93]: Shape information?

File: mmedit/models/backbones/vfi_backbones/flavr_net.py [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+@BACKBONES.register_module()
+class FLAVRNet(nn.Module):
+    """PyTorch implementation of FLAVR for video frame interpolation.
+
+    Paper:
+        FLAVR: Flow-Agnostic Video Representations for Fast Frame Interpolation
+    Ref repo: https://github.com/tarun005/FLAVR
+
+    Args:
+        input_frames (int): Number of input frames.
```
Comment (wangruohui) [inline mmedit/models/backbones/vfi_backbones/flavr_net.py:20]: `num_input_frames`, and `num_output_frames` are better because it is an integer number instead of a list of indices.

File: mmedit/models/backbones/vfi_backbones/flavr_net.py [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+        # normalization, Refer to https://github.com/myungsub/CAIN
+        mean_ = images.mean((2, 3, 4), keepdim=True)
+        images = images - mean_
+
+        xs = self.encoder(images)
+
+        dx_out = self.decoder(xs)
+
+        out = self.feature_fuse(dx_out)
+        out = self.conv_last(out)
```
Comment (wangruohui) [inline mmedit/models/backbones/vfi_backbones/flavr_net.py:94]: I meant we'd better add some comments here to indicate out's shape.

File: tests/test_models/test_backbones/test_vfi_backbones/test_flavr_net.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    # build model
+    model = build_backbone(model_cfg)
+
+    # test attributes
+    assert model.__class__.__name__ == 'FLAVRNet'
+
+    # prepare data
+    inputs = torch.rand(1, 4, 3, 256, 248)
+    target = torch.rand(1, 3, 256, 248)
```
Comment (wangruohui) [inline tests/test_models/test_backbones/test_vfi_backbones/test_flavr_net.py:30]: 248 or 256?

File: tests/test_models/test_backbones/test_vfi_backbones/test_flavr_net.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    with pytest.raises(OSError):
+        model.init_weights('')
+    with pytest.raises(TypeError):
+        model.init_weights(1)
+
+    model.decoder._join_tensors(inputs, inputs)
+
+    UpConv3d(4, 4, 1, 1, 1)
+    conv3d = UpConv3d(4, 4, 1, 1, 1, up_mode='trilinear', batchnorm=True)
+    conv3d(inputs)
```
Comment (wangruohui) [inline tests/test_models/test_backbones/test_vfi_backbones/test_flavr_net.py:46]: Any assertion for line 42-46?

**Review by wangruohui — DISMISSED** [review]

**Review by wangruohui — APPROVED** [review]
